erpnext/accounts/doctype/pos_invoice/pos_invoice.py

```diff
@@ -455,15 +461,32 @@ def get_existing_payment_request(self, pay):
 
 @frappe.whitelist()
 def get_stock_availability(item_code, warehouse):
+	bin_qty = get_bin_qty(item_code, warehouse)
+	pos_sales_qty = get_pos_reserved_qty(item_code, warehouse)
+
+	return bin_qty - pos_sales_qty
+
+def get_bundle_availability(bundle_item_code, warehouse):
+	product_bundle = frappe.get_doc('Product Bundle', bundle_item_code)
+
+	bundle_bin_qty = 1000000
+	for item in product_bundle.items:
+		item_bin_qty = get_bin_qty(item.item_code, warehouse)
+
+		max_num_of_bundles = item_bin_qty / item.qty
+		if bundle_bin_qty > max_num_of_bundles:
+			bundle_bin_qty = max_num_of_bundles
+
```

Comment:
```suggestion
	bundle_bin_qty = 1000000
	for item in product_bundle.items:
		sub_item_bin_qty = get_bin_qty(item.item_code, warehouse)
		sub_item_pos_reserved_qty = get_pos_reserved_qty(item.item_code, warehouse)
		
		available_qty = sub_item_bin_qty - sub_item_pos_reserved_qty
		
		max_available_bundles = available_qty / item.qty
		if bundle_bin_qty > max_available_bundles:
			bundle_bin_qty = max_available_bundles

```

Reply:
Can you check if this works? I just wanted to add pos reserved qty of sub-items since sub-items could be sold individually, but ended up renaming variables.

Reply:
yep, it works.